File: MG_thingQ/3rdWeek/printer.py
def solution(priorities, location):
    answer = 0
    idx_array = []
    value_array = priorities.copy()
    for i in range(len(priorities)):
        
        idx_array.append(i)
    
    i = 0 
    while True:
        if value_array[i]<max(value_array[i+1:]):
            idx_array.append(idx_array.pop(i))
            value_array.append(value_array.pop(i)) #index를 sort하는 것에 stack 사용.
        else :
            i +=1
        
        if value_array == sorted(value_array,reverse=True):
            break
    
    return idx_array.index(location)+1

File: SunshineJ/stack_queue/printer.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> priorities, int location) {
    int answer = 0;
    int max = 0, idx = 0, temp = 0;
    while(answer!=priorities.size()){
        for(int i=0;i<priorities.size();i++){
            int j = temp + i;
            if(j>=priorities.size()) j -= priorities.size();
            if(priorities[j]!=-1 && max<priorities[j]){
                    max = priorities[j];
                    idx = j;
            }
        }
        if(idx == location) return answer+1;
        else{
            priorities[idx] = -1;
            max = 0;
            answer++;
            temp = idx;
        }
    }
}
